src/main/java/org/broadleafcommerce/vendor/amazon/s3/S3FileServiceProvider.java

```diff
@@ -80,15 +90,42 @@ public File getResource(String name) {
 
     @Override
     public File getResource(String name, FileApplicationType fileApplicationType) {
-        File returnFile = blFileService.getLocalResource(buildResourceName(name));
+        File returnFile = blFileService.getLocalResource(getResourceName(name));
         OutputStream outputStream = null;
         InputStream inputStream = null;
 
         try {
             S3Configuration s3config = s3ConfigurationService.lookupS3Configuration();
             AmazonS3Client s3 = getAmazonS3Client(s3config);
-            S3Object object = s3.getObject(new GetObjectRequest(s3config.getDefaultBucketName(), buildResourceName(name)));
-            
+            String resourceName = getResourceName(name);
```

Comment:
That doesn't make sense to me. This is how I understand what you said: "Get the resource name as it is represented on the local file system. Now that we have the resource name as it is on the local filesystem, use that resource name to look it up on the remote file system (amazon)."

Why does a local resource name have relevance to the remote resource name?